### PR TITLE
feat(examples): Amazon Pay Button

### PR DESCRIPTION
**Pull request by mkucmus**
### Description

closes: #902 

see the [README.md](https://github.com/shopware/frontends/blob/81c32314ec5841d26e27bf1415d5c42266d53f93/examples/amazon-pay-button-example/README.md)

### CODE REVIEW

File: examples/amazon-pay-button-example/.vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "eslint.experimental.useFlatConfig": true
```
Comment (patzick) [inline examples/amazon-pay-button-example/.vscode/settings.json:2]: why is it here? I do not see any eslint flat config

File: examples/amazon-pay-button-example/README.md [hunk 2]
```diff
@@ -0,0 +1,188 @@
+# Toolset to manage Amazon Pay
+
+The example shows how to embed Amazon Pay button even without fully integrated backend (see 😇 **_Bypassing a plugin_** chapter in the bottom to check what it is about).
```
Comment (patzick) [inline examples/amazon-pay-button-example/README.md:3]: *_Bypassing a plugin_** - make it a link

File: examples/amazon-pay-button-example/README.md [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+```
+
+2. Setup nuxt project
+
+```ts
+// ./nuxt.config.ts
+export default defineNuxtConfig({
+  extends: ["@shopware-pwa/composables-next/nuxt-layer"],
+  modules: ["@shopware-pwa/nuxt3-module", "../src/module"], // path to amazon-pay module
+  amazonPay: {
```
Comment (patzick) [inline examples/amazon-pay-button-example/README.md:68]: shouldn't it be in the runtimeConfig?

File: examples/amazon-pay-button-example/package.json [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    "@nuxt/devtools": "^1.3.2",
+    "@nuxt/module-builder": "^0.7.0",
+    "@nuxt/schema": "^3.11.2",
+    "@shopware-pwa/composables-next": "canary",
+    "@shopware/api-client": "canary",
+    "@types/node": "^20.14.1",
+    "nuxt": "^3.11.2",
+    "typescript": "latest",
+    "vitest": "^1.6.0",
+    "vue-tsc": "^2.0.19"
```
Comment (patzick) [inline examples/amazon-pay-button-example/package.json:43]: please make all dependencies strict and make sure versions are the same with other versions in repo

File: examples/amazon-pay-button-example/playground/package.json [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+  "private": true,
+  "name": "my-module-playground",
+  "type": "module",
+  "scripts": {
+    "dev": "nuxi dev",
+    "build": "nuxi build",
+    "generate": "nuxi generate"
+  },
+  "dependencies": {
+    "nuxt": "^3.11.2",
```
Comment (patzick) [inline examples/amazon-pay-button-example/playground/package.json:11]: strict and matching dependency here as well

File: examples/amazon-pay-button-example/.npmrc [hunk 6]
```diff
@@ -0,0 +1,2 @@
+shamefully-hoist=true
+strict-peer-dependencies=false
```
Comment (patzick) [inline examples/amazon-pay-button-example/.npmrc:2]: why do we need this?

**Review by patzick — COMMENTED** [review]
Few suggestions from my side

File: examples/amazon-pay-button-example/.vscode/settings.json (hunk 1 above)
Comment (mkucmus) [reply examples/amazon-pay-button-example/.vscode/settings.json]: was part of nuxt module starter boilerplate, I removed it

File: examples/amazon-pay-button-example/README.md (hunk 3 above)
Comment (mkucmus) [reply examples/amazon-pay-button-example/README.md:67]: the value from module's config is copied to the runtime config in the module.

**Review by patzick — APPROVED** [review]

**Pull request merged**